woocommerce/woocommerce-ios
# Dynamic Dashboard: Last orders card UI

Closes: #12776
Part of: #12792


## Description


Starts showing last orders card in dashboard behind a feature flag. 

The work of loading the orders and applying the filters is partially completed. We still have to load the custom order statuses from storage. Pending work is tracked at #12792

Changes
- View and View model for the Last Order dashboard card. 
- Empty views for  the Last Order dashboard card. 
- Add an extension to provide background color based on order status.
- Synchronize orders based on selected status.
- Listen to storage and update the UI.
- Enable last orders card only if the store has orders. 

## Testing instructions


- Create a JN store with no orders
- Enable the feature flag `dynamicDashboardM2` and build and login into the app.
- From My store screen tap "Edit" to navigate to the customize screen.
- Validate that the "Most recent orders" card is marked as unavailable as there are no orders yet.
- Create orders using https://github.com/woocommerce/wc-smooth-generator
- Enable the "Most recent orders" card on the dashboard.
- Confirm that the card now displays the 3 most recent orders. If there are no orders available empty view should be displayed.
- Validate that the UI matches the design.
- Change the status filter and validate the orders are listed based on the selected status.
- If there are no orders available for a selected status empty view should be displayed.
- Disconnect the Internet connection and reload the card. Confirm that the error state is displayed.
- Tap the ellipsis button and dismiss the card. Confirm that event `dynamic_dashboard_hide_card_tapped` is tracked with type `last_orders`.

## Screenshots

| Dark | Light | Empty | Error | No orders at all |
|--------|--------|--------|--------|--------|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-21 at 12 52 27](https://github.com/woocommerce/woocommerce-ios/assets/524475/75998ee1-48fb-441f-b3af-6b97dc0270b4) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-21 at 12 52 01](https://github.com/woocommerce/woocommerce-ios/assets/524475/d6605fbb-4d3a-438a-900e-927694d8f774) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-21 at 08 30 38](https://github.com/woocommerce/woocommerce-ios/assets/524475/bdebf6ff-56e6-4a86-8d96-52defb9aaae9) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-21 at 08 30 26](https://github.com/woocommerce/woocommerce-ios/assets/524475/d5c7d57d-f7fa-4ea7-906f-7c7e2f68bbd3) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-21 at 14 26 57](https://github.com/woocommerce/woocommerce-ios/assets/524475/0d080285-3dcf-4e41-ae88-c0c1b7c99bf2) | 

---
